pkg/service/pipeline.go

```diff
@@ -1826,60 +1823,21 @@ func (s *service) ListPipelineRuns(ctx context.Context, req *pipelinepb.ListPipe
 
 		if CanViewPrivateData(run.Namespace, requesterUID) {
 			if len(run.Inputs) == 1 {
-				md, ok := metadataMap[run.Inputs[0].Name]
-				if !ok {
-					return nil, fmt.Errorf("failed to load input metadata. pipeline UID: %s input reference ID: %s", run.PipelineUID.String(), run.Inputs[0].Name)
-				}
-				pbRun.Inputs = make([]*structpb.Struct, 0)
-				err = json.Unmarshal(md, &pbRun.Inputs)
-				if err != nil {
-					return nil, err
-				}
-
+				key := run.Inputs[0].Name
+				pbRun.Inputs = parseMetadataToStructArray(metadataMap, log, key, "input",
+					zap.String("pipelineUID", run.PipelineUID.String()), zap.String("inputReferenceID", key))
```

Comment:
returned the error and logged in the function.
removed message for json unmarshal because it would be duplicated with the error msg provided by json package